xdsl/transforms/lower_riscv_func.py

```diff
@@ -64,15 +64,15 @@ def match_and_rewrite(self, op: riscv_func.SyscallOp, rewriter: PatternRewriter)
 class LowerRISCVFuncOp(RewritePattern):
     @op_type_rewrite_pattern
     def match_and_rewrite(self, op: riscv_func.FuncOp, rewriter: PatternRewriter):
-        body = op.func_body.block
-        first_op = body.first_op
+        first_block = op.func_body.blocks[0]
+        first_op = first_block.first_op
         assert first_op is not None
-        while len(body.args):
+        while len(first_block.args):
             # arguments are passed to riscv functions via a0, a1, ...
             # replace arguments with `GetRegisterOp`s
-            index = len(body.args) - 1
-            last_arg = body.args[-1]
-            get_reg_op = riscv.GetRegisterOp(f"a{index}")
+            index = len(first_block.args) - 1
+            last_arg = first_block.args[-1]
+            get_reg_op = riscv.GetRegisterOp(riscv.IntRegisterType(f"a{index}"))
```

Comment:
It seems does not consider the case when there are more than 8 function arguments?

Reply:
Not allowed by the RISC-V function calling convention

Reply:
My concern is whether the `riscv_func` allows more than 8 arguments and will rewrite auto spill the excess arguments on stacks.